### ISSUE TITLE
FileAlreadyExistsException when updating metadata

### ISSUE THREAD

**Issue by g4s8** [artipie/rpm-adapter#230]
RPM adapter failed once on testing stage with error:
```
java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: /tmp/6e705b584712528fbd486f1e39e76c5965b7e4c7788daf356d3ab00312e3a7ca-primary.xml.gz
```
And with logs:
```
[INFO] RxCachedThreadScheduler-31 [com.artipie.rpm.Rpm:infoForced 247] repository closed
[INFO] RxCachedThreadScheduler-31 [com.artipie.rpm.pkg.MetadataFile:infoForced 247] output com.artipie.rpm.pkg.FilelistsOutput@5e3afb8a closed
```
The update was started simultaneously from different threads (most probably).

The solution could be to throw an exception if multiple updates started simultaneously for one repository.

**Comment by 0crat** [comment]
The job [#230](https://github.com/artipie/rpm-adapter/issues/230) assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4e697089-bbab-4d06-afa6-bd95537fd838); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by olenagerasimova** [comment]
@g4s8 please take a look at #234, let's merge it and then I fix the bug.

**Comment by 0pdd** [comment]
@g4s8 2 puzzles [#244](https://github.com/artipie/rpm-adapter/issues/244), [#245](https://github.com/artipie/rpm-adapter/issues/245) are still not solved.

**Comment by 0crat** [comment]
@olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova) resigned from [#230](https://github.com/artipie/rpm-adapter/issues/230), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by olenagerasimova** [comment]
@0crat assign me

**Comment by olenagerasimova** [comment]
@g4s8 please close, we already have puzzles to continue

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/artipie/rpm-adapter/issues/230#issuecomment-644636185))

@olenagerasimova The job [#230](https://github.com/artipie/rpm-adapter/issues/230) assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/c516b385-de60-421d-a199-b6674a86ee70); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by g4s8** [comment]
@olenagerasimova thanks

**Comment by 0crat** [comment]
Job was finished in 0 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@69479bfe-25b6-466a-835a-a020547757e2, thread: PQ-CT2E6TK9B
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:artipie/rpm-adapter#230] declared for identity constraint of element "speed".
```

[0.53.15](https://github.com/zerocracy/farm): CID: [53420dd4-9a9d-468c-8a5e-1c8eaa8714d8](https://www.0crat.com/footprint/CT2E6TK9B/53420dd4-9a9d-468c-8a5e-1c8eaa8714d8), Type: "Order was finished"

**Comment by 0pdd** [comment]
@g4s8 all 2 puzzles are solved here: [#244](https://github.com/artipie/rpm-adapter/issues/244), [#245](https://github.com/artipie/rpm-adapter/issues/245).
